src/checkpoint/dispatch/dispatch.impl.h

```diff
@@ -155,19 +154,16 @@ buffer::ImplReturnType serializeType(T& to_serialize, BufferObtainFnType fn) {
 }
 
 template <typename T>
-T* deserializeType(
-  SerialByteType* data, SerialSizeType const& size, T* allocBuf
-) {
-  auto mem = allocBuf ?
-    reinterpret_cast<SerialByteType*>(allocBuf) : new SerialByteType[sizeof(T)];
-  auto& t = Dispatch<T>::unpackType(mem, data, size);
+T* deserializeType(SerialByteType* data, SerialByteType* allocBuf) {
+  auto mem = allocBuf ? allocBuf : new SerialByteType[sizeof(T)];
```

Comment:
Seems pedantic, but maybe important in the long run (i.e. once we're leaning on C++17 support): this needs to be `new (std::align_val_t{alignof T}) SerialByteType[sizeof(T)]`